src/Compilers/VisualBasic/Portable/Symbols/ConstraintsHelper.vb

```diff
@@ -1110,28 +1118,61 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return New CompoundDiagnosticInfo(diagnostics)
         End Function
 
+        Public Enum ConstructorConstraintError
+            None
+            NoPublicParameterlessConstructor
+            MustInheritType
+            HasRequiredMembers
+        End Enum
+
         ''' <summary>
         ''' Return true if the class type has a public parameterless constructor.
         ''' </summary>
-        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol) As Boolean
+        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol) As ConstructorConstraintError
             type = type.OriginalDefinition
-            Debug.Assert(type.TypeKind = TypeKind.Class)
+            Debug.Assert(type.TypeKind = TypeKind.Class OrElse type.TypeKind = TypeKind.Structure)
 
-            Dim sourceNamedType = TryCast(type, SourceNamedTypeSymbol)
+            Dim sourceClass = If(type.TypeKind = TypeKind.Class, TryCast(type, SourceNamedTypeSymbol), Nothing)
 
-            If sourceNamedType IsNot Nothing AndAlso Not sourceNamedType.MembersHaveBeenCreated Then
+            If sourceClass IsNot Nothing AndAlso Not sourceClass.MembersHaveBeenCreated Then
                 ' When we are dealing with group classes and synthetic entry points,
                 ' we can end up here while we are building the set of members for the type.
                 ' Using InstanceConstructors property will send us into an infinite loop.
-                Return sourceNamedType.InferFromSyntaxIfClassWillHavePublicParameterlessConstructor()
+                If sourceClass.InferFromSyntaxIfClassWillHavePublicParameterlessConstructor() Then
+                    Return ConstructorConstraintError.None
+                Else
+                    Return ConstructorConstraintError.NoPublicParameterlessConstructor
+                End If
             End If
 
+            Dim hasRequiredMembers = type.AllRequiredMembers.Count > 0
```

Comment:
>AllRequiredMembers [](http://example.com/codeflow?start=42&length=18)

Can we get an error sentinel here? Are we testing a scenario like that?

Reply:
`GenericConstrainedToNew_Forbidden_HasRequiredMembersError`